src/utilities/createRouterReject.ts

```diff
@@ -0,0 +1,91 @@
+import { Ref, ref } from 'vue'
+import { NotFound } from '@/components'
+import { RegisteredRejection, Resolved, Route, RouteComponent } from '@/types'
+
+export const builtInRejections = ['NotFound'] as const
+export type BuiltInRejection = typeof builtInRejections[number]
+
+export const builtInRejectionComponents: Record<BuiltInRejection, RouteComponent> = {
+  NotFound,
+}
+
+export type RouterRejection = BuiltInRejection | RegisteredRejection
+
+type BuiltInRejectionComponents = Partial<Record<BuiltInRejection, RouteComponent>>
+
+export type RouterRejectionComponents = RegisteredRejection extends never
+  ? { rejections?: BuiltInRejectionComponents }
+  : { rejections: BuiltInRejectionComponents & Record<RegisteredRejection, RouteComponent> }
+
+export type RouterReject = (type: RouterRejection) => void
+
+type GetRejectionComponent = (type: RouterRejection) => RouteComponent
+type GetRejectionRoute = (type: RouterRejection) => Resolved<Route>
+type ClearRejection = () => void
+export type RouterRejectionComponent = Ref<null | { type: RouterRejection, component: RouteComponent }>
```

Comment:
IMO this type name is a bit misleading since it's both the type and the component. Suggestion would be just `RouterRejection` and probably rename the existing `RouterRejection` to be `RouterRejectionType`?

Reply:
I struggled with this, and your suggestion seems so obvious 😂 will update!